pilot/cmd/pilot-agent/options/security.go

```diff
@@ -65,8 +65,11 @@ func NewSecurityOptions(proxyConfig *meshconfig.ProxyConfig, stsPort int, tokenM
 	var tokenManager security.TokenManager
 	if stsPort > 0 || xdsAuthProvider.Get() != "" {
 		// tokenManager is gcp token manager when using the default token manager plugin.
-		tokenManager = tokenmanager.CreateTokenManager(tokenManagerPlugin,
+		tokenManager, err = tokenmanager.CreateTokenManager(tokenManagerPlugin,
 			tokenmanager.Config{CredFetcher: o.CredFetcher, TrustDomain: o.TrustDomain})
+		if err != nil {
+			log.Errorf("failed to create token manager: %v", err)
```

Comment:
Why do we log here but return err below? Seems inconsistent

Reply:
The log here has been removed.